### PR TITLE
mupdf mupdf-tools 1.23.5 pymupdf 1.23.6

### PR DESCRIPTION
**Pull request by jonhermansen**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by p-linnane** [comment]
Please split this into 2 commits. One per formulae with the messages `mupdf 1.23.3` and `mupdf-tools 1.23.3`.

**Comment by jonhermansen** [comment]
> Please split this into 2 commits. One per formulae with the messages `mupdf 1.23.3` and `mupdf-tools 1.23.3`.

Thanks, I tried to follow the old changes and saw they were bumped in the same branch. But I didn't realize I should split the commits. It should be fixed now.

File: Formula/m/mupdf.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Mupdf < Formula
   desc "Lightweight PDF and XPS viewer"
   homepage "https://mupdf.com/"
-  url "https://mupdf.com/downloads/archive/mupdf-1.22.2-source.tar.gz"
-  sha256 "54c66af4e6ef8cea9867cc0320ef925d561b42919ea0d4f89db5c9ef485bbeb7"
+  url "https://mupdf.com/downloads/archive/mupdf-1.23.3-source.tar.gz"
+  sha256 "1ef9a6409bc0a3271586e1b16f78eb156a579521cd212a124b57c4da8b940aad"
   license "AGPL-3.0-or-later"
   revision 1
```
Comment (chenrui333) [inline Formula/m/mupdf.rb:7]:
```suggestion
```

Comment (jonhermansen) [reply Formula/m/mupdf.rb]: I rebased and removed this line from the mupdf commit.

**Comment by jonhermansen** [comment]
My bad, I did not see the Python dependency. Looking into the current build issue with it now...

**Comment by baskerville** [comment]
> My bad, I did not see the Python dependency. Looking into the current build issue with it now...

[This commit](https://github.com/ArtifexSoftware/mupdf/commit/3ffd5b7f529d951f97cb49bf271dcaf3be9c0635) explains the Python error:
```
ImportError: /home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.6/lib/python3.11/site-packages/fitz/_fitz.cpython-311-x86_64-linux-gnu.so: undefined symbol: pdf_lookup_anchor
```

**Comment by jonhermansen** [comment]
> > My bad, I did not see the Python dependency. Looking into the current build issue with it now...
> 
> [This commit](https://github.com/ArtifexSoftware/mupdf/commit/3ffd5b7f529d951f97cb49bf271dcaf3be9c0635) explains the Python error:
> ```
> ImportError: /home/linuxbrew/.linuxbrew/Cellar/python@3.11/3.11.6/lib/python3.11/site-packages/fitz/_fitz.cpython-311-x86_64-linux-gnu.so: undefined symbol: pdf_lookup_anchor
> ```

I tried fixing the Python lib issue but did not make any substantial progress. It looks like there were many changes to the project's setup.py file since it was last updated as well.

@baskerville If you have any ideas on how to proceed, I'm all ears. I was thinking to try a more incremental approach, going version by version first (or worse, commit by commit).

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @EricFromCanada @jonhermansen and everyone!

**Review by chenrui333 — COMMENTED** [review]
Nice work!

**Pull request merged**